analyzers/tests/SonarAnalyzer.TestFramework/MetadataReferences/CoreMetadataReference.cs

```diff
@@ -26,10 +26,6 @@ namespace SonarAnalyzer.TestFramework.MetadataReferences;
 
 public static class CoreMetadataReference
 {
-    public static MetadataReference MicrosoftExtensionsHostingAbstractions { get; } = Create(typeof(Microsoft.Extensions.Hosting.IHost));
-    public static MetadataReference MicrosoftExtensionsDependencyInjectionAbstractions { get; } = Create(typeof(Microsoft.Extensions.DependencyInjection.IServiceCollection));
-    public static MetadataReference MicrosoftExtensionsPrimitives { get; } = Create(typeof(Microsoft.Extensions.Primitives.StringValues));
```

Comment:
Why those 3 references were moved to AspNetCoreMetadataReference?
They are not technically part of ASP NET.

IMO, it is fine to let them be there.

Reply:
While they do not have AspNetCore in their name,, they are part of AspNetCore SDK, just like everything else in that file.

This cleanup also revealed a duplication because `MicrosoftExtensionsHostingAbstractions` was present in both places.